src/util/git.js

```diff
@@ -364,7 +365,7 @@ export default class Git implements GitRefResolvingInterface {
   }
 
   async setRefRemote(): Promise<string> {
-    const stdout = await spawnGit(['ls-remote', '--tags', '--heads', this.gitUrl.repository]);
+    const stdout = await spawnGit(['ls-remote', '--tags', '--heads', this.gitUrl.repository.split('#')[0]]);
```

Comment:
Why do we need this now?

Reply:
I do not know how to explain it but yarn attempts to run this command, which does not work:
`git ls-remote --tags --heads ssh://git@bitbucket.org/openagenda/validators#v1.12.2`
*instead of `git ls-remote --tags --heads ssh://git@bitbucket.org/openagenda/validators`*
(from https://github.com/yarnpkg/yarn/blob/master/src/resolvers/exotics/hosted-git-resolver.js#L239)

I can not reproduce this bug outside my big repo which gathers fifty repo that tangle each other, and use a lot of `validators`. This is the only way I have found to effectively target this bug.

Reply:
Alright, thanks for the explanation :)